drivers/at/at.c

```diff
@@ -479,6 +528,15 @@ ssize_t at_readline(at_dev_t *dev, char *resp_buf, size_t len, bool keep_eol, ui
 
             resp_pos += read_res;
             len -= read_res;
+
+            if (substr && (size_t)(resp_pos - resp_buf) >= substr_len) {
+                if (strncmp(substr_p, substr, substr_len) == 0) {
```

Comment:
Yes this could be implemented faster (hashes and so), but as long as the match is small, it doesn't matter and keeps everything nice and simple.
Currently, this is only used for prompt detection `>`, so I could make this even more simple and scan for a single character only. However, I thought this can be used for other stuff in the future. 
If someone doesn't like it I can replace it with a single-character variant.

Reply:
The whole point is to re-use it for other purposes in the future, otherwise I could have just checked for `>`. In fact, I will very probably use this function for echo detection in a next PR fixing URCs.

Reply:
I'd still like a comment here, it's not obvious what `substr` is and why this is done here.